From a218897361ae5c0e2176bbf2c854c2e82deb72a0 Mon Sep 17 00:00:00 2001
From: warped-rudi <r.ihle@s-t.de>
Date: Thu, 6 Aug 2015 08:33:58 +0200
Subject: [PATCH 4/5] video: mxc_hdmi: Add command line switch to disable EDID
 processing.

Allow the use of the fallback videomode list even in case a valid EDID
is obtainable. This is done by adding 'mxc_hdmi.ignore_edid=1' to the
kernel command line.
---
 drivers/video/mxc/mxc_hdmi.c |   20 ++++++++++++++------
 1 file changed, 14 insertions(+), 6 deletions(-)

diff --git a/drivers/video/mxc/mxc_hdmi.c b/drivers/video/mxc/mxc_hdmi.c
index 75b0a1b..ab9625f 100644
--- a/drivers/video/mxc/mxc_hdmi.c
+++ b/drivers/video/mxc/mxc_hdmi.c
@@ -214,6 +214,10 @@ static char *rgb_quant_range = "auto";
 module_param(rgb_quant_range, charp, S_IRUGO);
 MODULE_PARM_DESC(rgb_quant_range, "RGB Quant Range (auto, default, limited, full)");
 
+static bool ignore_edid = 0;
+module_param(ignore_edid, bool, S_IRUGO);
+MODULE_PARM_DESC(ignore_edid, "Ignore EDID (default=0)");
+
 static struct platform_device_id imx_hdmi_devtype[] = {
 	{
 		.name = "hdmi-imx6DL",
@@ -2022,13 +2026,17 @@ static void mxc_hdmi_cable_connected(struct mxc_hdmi *hdmi)
 	hdmi->cable_plugin = true;
 
 	/* HDMI Initialization Step C */
-	hdmi->edid_status = mxc_hdmi_read_edid(hdmi);
-
-	/* Read EDID again if first EDID read failed */
-	if (hdmi->edid_status == HDMI_EDID_NO_MODES ||
-			hdmi->edid_status == HDMI_EDID_FAIL) {
-		dev_info(&hdmi->pdev->dev, "Read EDID again\n");
+	if (ignore_edid) {
+		hdmi->edid_status = HDMI_EDID_FAIL;
+	} else {
 		hdmi->edid_status = mxc_hdmi_read_edid(hdmi);
+
+		/* Read EDID again if first EDID read failed */
+		if (hdmi->edid_status == HDMI_EDID_NO_MODES ||
+				hdmi->edid_status == HDMI_EDID_FAIL) {
+			dev_info(&hdmi->pdev->dev, "Read EDID again\n");
+			hdmi->edid_status = mxc_hdmi_read_edid(hdmi);
+		}
 	}
 
 	/* HDMI Initialization Steps D, E, F */
-- 
1.7.9.5

